store/index-refactor.js

```diff
@@ -58,5 +70,12 @@ export const mutations = {
             p_state.columnToCategoryMapping[columnName] = targetCategory;
         }
 
+    },
+
+    createColumnToCategoryMap(p_state) {
+
+        // Column to category map lists all columns as keys with default value of null
+        p_state.columnToCategoryMapping =
+            Object.fromEntries(p_state.getColumnNames().map((column) => [column, null]));
```

Comment:
`getColumnNames` is a getter inside `store.getters...`, you can't reach it in the `store.state`. Mutations get the `store.state` object as their first argument, but not the whole `store`. That is by design although I don't fully get the reasoning (see https://www.github.com/vuejs/vuex/issues/684). I think there are two options here:

1. Have the mutation expect an argument with the list of columnNames. So far the only place we'll call this from is the (future) `action` that is triggered by uploading a new dataTable. So it would make sense for this action to provide the list of category names in the payload. And that would also remove the need to dependence on `dataTable` already being accessible in the `state` - which is a bit odd.
2. Turn this thing into an action instead. Actions "receive a context object which exposes the same set of methods/properties on the store instance", so you can do the whole `context.getters...` thing (see https://vuex.vuejs.org/guide/actions.html).

Between the two, I am much in favour of 1. because it seems a lot cleaner to me. But let me know what you think.

Reply:
We went with option 1 after meeting between myself, @surchs, and @rmanaem.